torch_memory_saver/__init__.py

```diff
@@ -59,7 +58,10 @@ def enabled(self):
     def compute():
         env_ld_preload = os.environ.get('LD_PRELOAD', '')
         if 'torch_memory_saver' in env_ld_preload:
-            return _BinaryInfo(cdll=ctypes.CDLL(env_ld_preload))
+            paths = env_ld_preload.split(':')
+            lib_path = next((path for path in paths if 'torch_memory_saver' in path), None)
+            if lib_path:
```

Comment:
nit: looks like we may have a bug for the else case (or, if the `else` branch will never occur, we may change if to assert)